modules/Expectation.js

```diff
@@ -440,14 +440,35 @@ class Expectation {
     )
 
     assert(
-      spy.calls.some(call => isEqual(call.arguments, expectedArgs)),
+      spy.calls.some(call => call.arguments.some((argument, index) =>
+        argument === call.arguments[index]
+      )),
```

Comment:
This is just a naive implementation, not even sure of it works. From what I have seen it is probably necessary to scale this upstream for the [`is-equal`](https://www.npmjs.com/package/is-equal) package because it seems that it doesn't differentiate instances comparison on the `is-equal/why` submodule.